craco.config.cjs

```diff
@@ -29,25 +32,30 @@ module.exports = {
     },
   },
   webpack: {
-    plugins: [
-      new VanillaExtractPlugin({ identifiers: 'short' }),
-      new DefinePlugin({
-        'process.env.REACT_APP_GIT_COMMIT_HASH': JSON.stringify(commitHash.toString()),
-      }),
-    ],
+    plugins: [new VanillaExtractPlugin({ identifiers: 'short' })],
     configure: (webpackConfig) => {
-      const instanceOfMiniCssExtractPlugin = webpackConfig.plugins.find(
-        (plugin) => plugin instanceof MiniCssExtractPlugin
-      )
-      if (instanceOfMiniCssExtractPlugin !== undefined) instanceOfMiniCssExtractPlugin.options.ignoreOrder = true
+      // Extend process.env with dynamic values (eg commit hash).
+      Object.assign(webpackConfig.plugins.find((plugin) => plugin instanceof DefinePlugin).definitions['process.env'], {
+        REACT_APP_GIT_COMMIT_HASH: JSON.stringify(commitHash),
+      })
+
+      if (isProduction) {
+        // Type checking and linting are only necessary as part of development and testing.
+        // Omit them from production builds, as they slow down the feedback loop.
+        webpackConfig.plugins = webpackConfig.plugins.filter((plugin) => {
+          if (plugin instanceof ForkTsCheckerWebpackPlugin) return false
+          if (plugin instanceof EsLintWebpackPlugin) return false
+          return true
+        })
+
+        // CSS ordering is mitigated through scoping / naming conventions, so we can ignore order warnings.
+        // See https://webpack.js.org/plugins/mini-css-extract-plugin/#remove-order-warnings.
+        webpackConfig.plugins.find((plugin) => plugin instanceof MiniCssExtractPlugin).options.ignoreOrder = true
```

Comment:
do we lose our undefined check here on purpose?

Reply:
Yes: I've moved this into the `isProduction` check - it is always defined in production environments.

Reply:
i'm okay with this but i think it's not necessarily clear that those two things are related - i would interpret this as only setting `ignoreOrder` in a production environment as a future reader, not that find will always be successful only in a prod environment